### PR TITLE
Add jit helper to detect `@DontInline` annotation

### PR DESCRIPTION
**Pull request by EricYangIBM**
Closes: #13237
Signed-off-by: Eric Yang <eric.yang@ibm.com>

### CODE REVIEW

**Comment by EricYangIBM** [comment]
Tested by explicitly calling jitIsMethodTaggedWithDontInline in annhelp_tests.c and substituting `ForceInline` with `DontInline` in ContainsRuntimeAnnotationTest.java

**Comment by tajila** [comment]
Can you also add a test case

**Comment by EricYangIBM** [comment]
> Can you also add a test case

Done

File: test/functional/Java8andUp/src_110_up/org/openj9/test/annotation/ContainsRuntimeAnnotationTest.java [hunk 1]
```diff
@@ -280,6 +284,16 @@ public void test_overridden_methods() throws Exception {
 		Assert.assertTrue(annotationFound, "I.method3 doesn't have MyMethodAnnotation annotation");
 	}
 
+	@Test
+	public void test_dont_inline() throws Exception {
```
Comment (tajila) [inline test/functional/Java8andUp/src_110_up/org/openj9/test/annotation/ContainsRuntimeAnnotationTest.java:288]: can you also add the src_80 version

Comment (EricYangIBM) [reply test/functional/Java8andUp/src_110_up/org/openj9/test/annotation/ContainsRuntimeAnnotationTest.java:288]: I thought you couldn't import ForceInline and DontInline in java8

Comment (tajila) [reply test/functional/Java8andUp/src_110_up/org/openj9/test/annotation/ContainsRuntimeAnnotationTest.java:288]: ah right, I forgot

**Review by tajila — APPROVED** [review]

**Comment by tajila** [comment]
Jenkins test sanity,extended plinux jdk8,jdk11

**Comment by tajila** [comment]
@0xdaryl ^^

**Pull request merged**